From bc998c6600c1557b40a11096f17a176e182a27c2 Mon Sep 17 00:00:00 2001
From: Assam Boudjelthia <assam.boudjelthia@qt.io>
Date: Mon, 29 Sep 2025 13:05:00 +0300
Subject: [PATCH 233/553] Fix Android builds with QT_FORCE_BUILD_TOOLS=ON

Bootstrapped builds don't have streamable qCDebug() calls, so use printf
style.

Also, QJniObject can't be used in this build type, so guard it under
QCoreApplicationPrivate::appVersion().

Fixes: QTBUG-140516
Pick-to: 6.8
Change-Id: I912e57689f758627ef6ca63405b557940adbf186
Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>
(cherry picked from commit 039677408eb1acb6bf1c5afcdfcda7be93e67f17)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/global/qnativeinterface_p.h | 2 +-
 src/corelib/kernel/qcoreapplication.cpp | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/corelib/global/qnativeinterface_p.h b/src/corelib/global/qnativeinterface_p.h
index 93807f0ba67..6f90bccb46e 100644
--- a/src/corelib/global/qnativeinterface_p.h
+++ b/src/corelib/global/qnativeinterface_p.h
@@ -41,7 +41,7 @@ QT_DECLARE_EXPORTED_QT_LOGGING_CATEGORY(lcNativeInterface, Q_CORE_EXPORT)
                     "Match for interface %s. Comparing revisions (requested %d / available %d)",   \
                     name, revision, TypeInfo<NativeInterface>::revision());                        \
             if (revision == TypeInfo<NativeInterface>::revision()) {                               \
-                qCDebug(lcNativeInterface) << "Full match. Returning dynamic cast of" << baseType; \
+                qCDebug(lcNativeInterface, "Full match. Returning dynamic cast of %p", baseType);  \
                 return dynamic_cast<NativeInterface *>(baseType);                                  \
             } else {                                                                               \
                 qCWarning(lcNativeInterface,                                                       \
diff --git a/src/corelib/kernel/qcoreapplication.cpp b/src/corelib/kernel/qcoreapplication.cpp
index 6da2f63fe16..80a8544c933 100644
--- a/src/corelib/kernel/qcoreapplication.cpp
+++ b/src/corelib/kernel/qcoreapplication.cpp
@@ -193,7 +193,7 @@ QString QCoreApplicationPrivate::appVersion() const
     QString applicationVersion;
 #if defined(Q_OS_DARWIN)
     applicationVersion = infoDictionaryStringProperty(QStringLiteral("CFBundleVersion"));
-#elif defined(Q_OS_ANDROID)
+#elif defined(Q_OS_ANDROID) && !defined(QT_BOOTSTRAPPED)
     QJniObject context(QNativeInterface::QAndroidApplication::context());
     if (context.isValid()) {
         QJniObject pm = context.callObjectMethod(
@@ -814,7 +814,7 @@ void Q_TRACE_INSTRUMENT(qtcore) QCoreApplicationPrivate::init()
     if (!coreappdata()->applicationVersionSet)
         coreappdata()->applicationVersion = appVersion();
 
-#if defined(Q_OS_ANDROID)
+#if defined(Q_OS_ANDROID) && !defined(QT_BOOTSTRAPPED)
     // We've deferred initializing the logging registry due to not being
     // able to guarantee that logging happened on the same thread as the
     // Qt main thread, but now that the Qt main thread is set up, we can
-- 
2.51.2

